### PR TITLE
portmod: init at 2.0_beta7

### PR DESCRIPTION
**Pull request by marius851000**
###### Motivation for this change
wanted to add the portmod package. I use it, and wanted it to be in the nixpkgs.

###### Things done

mainly in additional to portmod, add patool.

I was able to install the meta-momw/expanded-vanilla , but with some modification in the environment: needed to set "TMPDIR=/tmp" (the default temp dir doesn't have enought size) and to have unrar in the PATH.

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by IvarWithoutBones** [comment]
Result of `nixpkgs-review pr 98093` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>python27Packages.redbaron</li>
  </ul>
</details>
<details>
  <summary>15 packages built:</summary>
  <ul>
    <li>patool (python38Packages.patool)</li>
    <li>portmod</li>
    <li>python27Packages.baron</li>
    <li>python27Packages.patool</li>
    <li>python27Packages.pypblib</li>
    <li>python27Packages.python-sat</li>
    <li>python37Packages.baron</li>
    <li>python37Packages.patool</li>
    <li>python37Packages.pypblib</li>
    <li>python37Packages.python-sat</li>
    <li>python37Packages.redbaron</li>
    <li>python38Packages.baron</li>
    <li>python38Packages.pypblib</li>
    <li>python38Packages.python-sat</li>
    <li>python38Packages.redbaron</li>
  </ul>
</details>

`python27Packages.redbaron` error: 
```
[2/11/31 built, 149 copied (1588.1/1588.6 MiB), 285.6 MiB DL] building python3.7-python-sat-0.1.6.dev6 (buildPhase): |       ^~~
builder for '/nix/store/1pi10m6myd6rcjw7k5iz9nxwm16nzsdr-python2.7-redbaron-0.9.2.drv' failed with exit code 2; last 10 log lines:
  HINT: remove __pycache__ / .pyc files and/or use a unique basename for your test file modules
  ____________________ ERROR collecting tests/test_setter.py _____________________
  import file mismatch:
  imported module 'tests.test_setter' has this __file__ attribute:
    /home/psycojoker/code/python/redbaron/tests/test_setter.py
  which is not the same as the test file we want to collect:
    /build/redbaron-0.9.2/tests/test_setter.py
  HINT: remove __pycache__ / .pyc files and/or use a unique basename for your test file modules
  !!!!!!!!!!!!!!!!!!! Interrupted: 13 errors during collection !!!!!!!!!!!!!!!!!!!
  =========================== 13 error in 8.46 seconds ===========================
cannot build derivation '/nix/store/jx4pgvd31fy01ldzvplk4kg8qizcf7xh-env.drv': 1 dependencies couldn't be built
[29 built (1 failed), 149 copied (1588.1 MiB), 285.6 MiB DL]
error: build of '/nix/store/jx4pgvd31fy01ldzvplk4kg8qizcf7xh-env.drv' failed
```

**Comment by marius851000** [comment]
that should be fixed, thank.

**Review by IvarWithoutBones — APPROVED** [review]
Result of `nixpkgs-review pr 98093` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>16 packages built:</summary>
  <ul>
    <li>patool (python38Packages.patool)</li>
    <li>portmod</li>
    <li>python27Packages.baron</li>
    <li>python27Packages.patool</li>
    <li>python27Packages.pypblib</li>
    <li>python27Packages.python-sat</li>
    <li>python27Packages.redbaron</li>
    <li>python37Packages.baron</li>
    <li>python37Packages.patool</li>
    <li>python37Packages.pypblib</li>
    <li>python37Packages.python-sat</li>
    <li>python37Packages.redbaron</li>
    <li>python38Packages.baron</li>
    <li>python38Packages.pypblib</li>
    <li>python38Packages.python-sat</li>
    <li>python38Packages.redbaron</li>
  </ul>
</details>

File: pkgs/development/python-modules/baron/default.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  version = "0.9";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sFifMrkc9txFDqanG0oijEM++HVrQfq/iIFaPC05Kzo=";
+  };
+
+  propagatedBuildInputs = [ rply ];
+
+  checkInputs = [ pytest ];
```
Comment (jonringer) [inline pkgs/development/python-modules/baron/default.nix:14]:
```suggestion
  checkInputs = [ pytestCheckHook ];
```

File: pkgs/development/python-modules/patool/default.nix [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+buildPythonApplication rec {
+  pname = "patool";
+  version = "1.12";
+
+  #pypi doesn't have test data
+  src = fetchFromGitHub {
+    owner = "wummel";
+    repo = pname;
+    rev = "upstream/${version}";
+    sha256 = "Xv4aCUnLi+b1T29tuKRADTIWwK2dO8iDP/D7UfU5mWw=";
```
Comment (jonringer) [inline pkgs/development/python-modules/patool/default.nix:34]: There's still a preference to having base32 hashes within nixpkgs to ease the upgrade process for users with holder versions of the nix cli.

You can use `nix-hash --to-base32 <hash>` to convert the hash into the related base32 hash.

For more details refer to: https://github.com/NixOS/nixpkgs/pull/89308#issuecomment-636927054 #89275

File: pkgs/development/python-modules/pypblib/default.nix [hunk 3]
```diff
@@ -0,0 +1,18 @@
+{ lib, buildPythonPackage, fetchPypi }:
+
+buildPythonPackage rec {
+	pname = "pypblib";
+	version = "0.0.4";
+
+	src = fetchPypi {
+		inherit pname version;
+		sha256 = "cd2TC/F3yjjW7rRzcC0F3wfhHyA4LbDHZkZSl+r0kGI=";
```
Comment (jonringer) [inline pkgs/development/python-modules/pypblib/default.nix:9]: There's still a preference to having base32 hashes within nixpkgs to ease the upgrade process for users with holder versions of the nix cli.

You can use `nix-hash --to-base32 <hash>` to convert the hash into the related base32 hash.

For more details refer to: https://github.com/NixOS/nixpkgs/pull/89308#issuecomment-636927054 #89275

File: pkgs/development/python-modules/python-sat/default.nix [hunk 4]
```diff
@@ -0,0 +1,22 @@
+{ buildPythonPackage, fetchPypi, lib, six, pypblib, pytest }:
+
+buildPythonPackage rec {
+	pname = "python-sat";
+  version = "0.1.6.dev6";
+
+  src = fetchPypi {
+		inherit pname version;
+    sha256 = "7v8IP0JK3Xjmf4uuY/Z25aRINU15A2H0hNGFI8E5Or0=";
```
Comment (jonringer) [inline pkgs/development/python-modules/python-sat/default.nix:9]: There's still a preference to having base32 hashes within nixpkgs to ease the upgrade process for users with holder versions of the nix cli.

You can use `nix-hash --to-base32 <hash>` to convert the hash into the related base32 hash.

For more details refer to: https://github.com/NixOS/nixpkgs/pull/89308#issuecomment-636927054 #89275

File: pkgs/development/python-modules/redbaron/default.nix [hunk 5]
```diff
@@ -0,0 +1,27 @@
+{ lib, fetchPypi, buildPythonPackage, baron, pytest }:
+
+buildPythonPackage rec {
+  pname = "redbaron";
+  version = "0.9.2";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "Ry0HOcprIkC7IniuQoYEp1RyycEuhsYyHowBYTnAEy8=";
```
Comment (jonringer) [inline pkgs/development/python-modules/redbaron/default.nix:9]: There's still a preference to having base32 hashes within nixpkgs to ease the upgrade process for users with holder versions of the nix cli.

You can use `nix-hash --to-base32 <hash>` to convert the hash into the related base32 hash.

For more details refer to: https://github.com/NixOS/nixpkgs/pull/89308#issuecomment-636927054 #89275

File: pkgs/top-level/all-packages.nix [hunk 6]
```diff
@@ -2298,6 +2298,8 @@ in
 
   patdiff = callPackage ../tools/misc/patdiff { };
 
+  patool = python3Packages.patool;
```
Comment (jonringer) [inline pkgs/top-level/all-packages.nix:2301]: only python applications should be exported in all-packages
```suggestion
  patool = with python3Packages; toPythonApplication patool;
```

**Comment by jonringer** [comment]
I take back the hash reviews, apparently master has been updated with a minver which supports 64bit base hashes

**Comment by marius851000** [comment]
Thank. I fixed the issue you reported (exluding the hash). Now I know why hash are shorter now.

File: pkgs/development/python-modules/pypblib/default.nix [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+	};
+
```
Comment (jonringer) [inline pkgs/development/python-modules/pypblib/default.nix:11]: There are no tests included. Please try to checkout from source and check if they have unit tests, and try to run them. Unit tests give a good indication that they package has a high degree of validity and correctness given the python package set.

If tests are not available, then please use `pythonImportsCheck` to import the most important modules. This isn't as good as unit tests, but will usually give a good indication of run-time errors. Please see [pythonImportsCheck documentation](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pythonimportscheck) for more information.

```

----------------------------------------------------------------------
Ran 0 tests in 0.000s
```

File: pkgs/development/python-modules/python-sat/default.nix [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+  };
+
+	propagatedBuildInputs = [ six pypblib ];
+
+	doCheck = false; # test are interactive
```
Comment (jonringer) [inline pkgs/development/python-modules/python-sat/default.nix:14]: If tests are not available, then please use `pythonImportsCheck` to import the most important modules. This isn't as good as unit tests, but will usually give a good indication of run-time errors. Please see [pythonImportsCheck documentation](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pythonimportscheck) for more information.

Comment (marius851000) [reply pkgs/development/python-modules/python-sat/default.nix]: tests weren't included on pypi, but they were on github

**Comment by marius851000** [comment]
That's done. Thank again.

**Comment by doronbehar** [comment]
There's an editor config issue.

**Comment by marius851000** [comment]
Thank. Didn't noticed this.

File: pkgs/development/python-modules/python-sat/default.nix [hunk 9]
```diff
@@ -0,0 +1,24 @@
+{ buildPythonPackage, fetchFromGitHub, lib, six, pypblib, pytestCheckHook }:
+
+buildPythonPackage rec {
+	pname = "python-sat";
+  version = "0.1.6.dev6";
```
Comment (doronbehar) [inline pkgs/development/python-modules/python-sat/default.nix:5]: EditorConfig didn't catch that, but there's mixed indentation in this file. Please use spaces instead of tabs and 2 of them for 1 level of indent.

File: pkgs/development/python-modules/redbaron/default.nix [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+  };
+
+  propagatedBuildInputs = [ baron ];
+
+  checkPhase = ''
+    rm -rf tests/__pycache__
+    pytest --ignore=tests/test_bounding_box.py
+  ''; #error about fixtures
+
+  checkInputs = [ pytest ];
```
Comment (doronbehar) [inline pkgs/development/python-modules/redbaron/default.nix:19]: I think there's a better way to do that, with `pytestCheckHook` as a start, and `disabledTests`.

File: pkgs/games/portmod/tr-patcher.nix [hunk 11]
```diff
@@ -0,0 +1,28 @@
+{ stdenv, fetchurl, fetchzip, jre, makeWrapper }:
+let
+	translation-file = fetchurl {
```
Comment (doronbehar) [inline pkgs/games/portmod/tr-patcher.nix:3]: `s/\t/  /g`.

File: pkgs/games/portmod/default.nix [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+      --deselect test/test_loader.py::test_execute_permissions_bleed \
+      --deselect test/test_loader.py::test_git
+  '';
+
+  # for some reason, installPhase doesn't copy the compiled binary
+  postInstall = ''
+    cp ${portmod-rust}/lib/libportmod.so $(echo $out/lib/python*/*/portmod)/portmod.so
+		for script in $out/bin/*
+		do
+		  wrapProgram $script \
```
Comment (doronbehar) [inline pkgs/games/portmod/default.nix:97]: Please use `makeWrapperArgs` to spare double wrapping.

File: pkgs/games/portmod/default.nix [hunk 13]
```diff
@@ -0,0 +1,109 @@
...
+    pname = "portmod-rust";
+
+    cargoSha256 = "P6OMRaOiMDrfDZBs7ofnoaMkB0q9SME6K3qwJSPnDhk=";
+
+    nativeBuildInputs = [ python3Packages.python ];
+
+    doCheck = false;
+  };
+
+  bin-program = [
```
Comment (doronbehar) [inline pkgs/games/portmod/default.nix:26]:
```suggestion
  bin-programs = [
```

File: pkgs/games/portmod/default.nix [hunk 14]
```diff
@@ -0,0 +1,109 @@
...
+  checkPhase = ''
+    cp ${portmod-rust}/lib/libportmod.so portmod/portmod.so
+    export HOME=$(mktemp -d)
+    pytest \
+      --deselect test/test_masters.py::test_masters_esp \
+      --deselect test/test_log.py::test_logging \
+      --deselect test/test_loader.py::test_execute_network_permissions \
+      --deselect test/test_loader.py::test_execute_permissions_bleed \
+      --deselect test/test_loader.py::test_git
+  '';
```
Comment (doronbehar) [inline pkgs/games/portmod/default.nix:90]: Same argument as before for pytestCheckHook.

File: pkgs/games/portmod/default.nix [hunk 15]
```diff
@@ -0,0 +1,109 @@
...
+		do
+		  wrapProgram $script \
+        --prefix GIT_SSL_CAINFO : ${cacert}/etc/ssl/certs/ca-bundle.crt \
+		    --prefix PATH : ${lib.makeBinPath bin-program }
+		done
+  '';
+
+  meta = {
+    description = "mod manager for openMW based on portage";
+    homepage = "https://gitlab.com/portmod/portmod/-/tree/master";
```
Comment (doronbehar) [inline pkgs/games/portmod/default.nix:105]:
```suggestion
    homepage = "https://gitlab.com/portmod/portmod";
```

File: pkgs/development/python-modules/python-sat/default.nix [hunk 16]
```diff
@@ -0,0 +1,24 @@
...
+
+  src = fetchFromGitHub {
+		owner = "pysathq";
+		repo = "pysat";
+		rev = version;
+    sha256 = "3hgZG9emcjWYrIUOoYHH2ktDAu6/o8frPdz/PTHsk70=";
```
Comment (doronbehar) [inline pkgs/development/python-modules/python-sat/default.nix:11]: Same argument as jonringer's regarding base32.

File: pkgs/development/python-modules/patool/default.nix [hunk 17]
```diff
@@ -0,0 +1,60 @@
...
+
+  checkInputs = [ pytest ] ++ compression-utility;
+
+  # for some reason, zip can't decompress file that doesn't finish by .zip, but patool doesn't take this in account
+  checkPhase = ''
+    pytest --deselect tests/archives/test_unzip.py::TestUnzip::test_unzip \
+      --deselect tests/archives/test_unzip.py::TestUnzip::test_unzip_file \
+      --deselect tests/archives/test_zip.py::TestZip::test_zip \
+      --deselect tests/archives/test_zip.py::TestZip::test_zip_file
+  '';
```
Comment (doronbehar) [inline pkgs/development/python-modules/patool/default.nix:52]: Same argument as before regarding pytestCheckHook.

File: pkgs/development/python-modules/patool/default.nix [hunk 18]
```diff
@@ -0,0 +1,60 @@
+{ stdenv, lib, buildPythonPackage, fetchFromGitHub, pytest, p7zip, unzip,
+  cabextract, zip, zopfli, lzip, zpaq, gnutar, gnugrep, diffutils, file,
+  gzip, bzip2, xz}:
+
+# unrar is unfree, as well as 7z with unrar support, not including it (patool doesn't support unar)
+
+let
+  compression-utility = [
```
Comment (doronbehar) [inline pkgs/development/python-modules/patool/default.nix:8]:
```suggestion
  compression-utilities = [
```

**Review by doronbehar — CHANGES_REQUESTED** [review]

File: pkgs/games/portmod/tr-patcher.nix (hunk 11 above)
Comment (marius851000) [reply pkgs/games/portmod/tr-patcher.nix:3]: I don't understand. Is that typing error or a sed script ?

Comment (doronbehar) [reply pkgs/games/portmod/tr-patcher.nix:3]: It's what ever you want :). Use whatever tool you wish, just be consistent please with spaces.

Comment (SuperSandro2000) [reply pkgs/games/portmod/tr-patcher.nix:3]: @marius851000 That is a regex. ``\t`` is a tab and you should replace all tabs with spaces.

Comment (marius851000) [reply pkgs/games/portmod/tr-patcher.nix:3]: Thank for this explanation. I'm not used to sed.

File: pkgs/development/python-modules/pypblib/default.nix [hunk 19]
```diff
@@ -0,0 +1,20 @@
+{ lib, buildPythonPackage, fetchPypi }:
+
+buildPythonPackage rec {
+	pname = "pypblib";
```
Comment (doronbehar) [inline pkgs/development/python-modules/pypblib/default.nix:4]: Tabs. Use 2 spaces.

File: pkgs/development/python-modules/patool/default.nix [hunk 20]
```diff
@@ -0,0 +1,58 @@
...
+    rev = "upstream/${version}";
+    sha256 = "0v4r77sm3yzh7y1whfwxmp01cchd82jbhvbg9zsyd2yb944imzjy";
+  };
+
+  prePatch = ''
+    substituteInPlace patoolib/util.py \
+      --replace "path = None" 'path = append_to_path(os.environ["PATH"], "${lib.makeBinPath compression-utilities}")'
+  '';
+
+  checkInputs = [ pytest ] ++ compression-utilities;
```
Comment (doronbehar) [inline pkgs/development/python-modules/patool/default.nix:42]:
```suggestion
  checkInputs = [ pytestCheckHook ] ++ compression-utilities;
```

File: pkgs/development/python-modules/patool/default.nix [hunk 21]
```diff
@@ -0,0 +1,58 @@
...
+
+  # for some reason, zip can't decompress file that doesn't finish by .zip, but patool doesn't take this in account
+  checkPhase = ''
```
Comment (doronbehar) [inline pkgs/development/python-modules/patool/default.nix:45]: Per https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/explicit-phases.md 

```suggestion
  preCheck = ''
```

File: pkgs/games/portmod/default.nix [hunk 22]
```diff
@@ -0,0 +1,98 @@
...
+
+    doCheck = false;
+  };
+
+  bin-programs = [
+    bubblewrap
+    git
+    python3Packages.virtualenv
+    (callPackage ./tes3cmd.nix { })
+    (callPackage ./tr-patcher.nix { })
```
Comment (doronbehar) [inline pkgs/games/portmod/default.nix:31]: I think those would be worth something if exposed in `all-packages.nix` (And be added in separate commits of their own).

File: pkgs/development/python-modules/patool/default.nix [hunk 23]
```diff
@@ -0,0 +1,58 @@
...
+    pytest --deselect tests/archives/test_unzip.py::TestUnzip::test_unzip \
```
Comment (doronbehar) [inline pkgs/development/python-modules/patool/default.nix:46]: Doesn't `disabeldTests` work?

File: pkgs/development/python-modules/redbaron/default.nix [hunk 24]
```diff
@@ -0,0 +1,27 @@
...
+    inherit pname version;
+    sha256 = "0bqkq0wn20cc3qrcd1ifq74p4m570j345bkq4axl08kbr8whfba7";
+  };
+
+  propagatedBuildInputs = [ baron ];
+
+  preCheck = ''
+    rm -rf tests/__pycache__
+    rm tests/test_bounding_box.py
+  ''; #error about fixtures
```
Comment (doronbehar) [inline pkgs/development/python-modules/redbaron/default.nix:17]: Can't use `disabledTests`?

**Review by doronbehar — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/redbaron/default.nix (hunk 24 above)
Comment (marius851000) [reply pkgs/development/python-modules/redbaron/default.nix:17]: No, I need to use the --ignore flag, as it is the parsing of the python file that cause problem (removing it also work).

File: pkgs/games/portmod/default.nix (hunk 22 above)
Comment (marius851000) [reply pkgs/games/portmod/default.nix:31]: Should I put them into different folder, or should I import call the file that are in the portmod subdir ?

File: pkgs/development/python-modules/python-sat/default.nix [hunk 25]
```diff
@@ -0,0 +1,24 @@
+{ buildPythonPackage, fetchFromGitHub, lib, six, pypblib, pytestCheckHook }:
+
+buildPythonPackage rec {
+	pname = "python-sat";
```
Comment (doronbehar) [inline pkgs/development/python-modules/python-sat/default.nix:4]: Do you differentiate between spaces and tabs in your text editor?

File: pkgs/games/tr-patcher/default.nix [hunk 26]
```diff
@@ -0,0 +1,28 @@
...
+    url = "https://gitlab.com/bmwinger/tr-patcher/uploads/b57899980b2351c136393f02977c4fab/tr-patcher-shadow.zip";
+    sha256 = "0va7nbmlgf3p2nc0z2b9n1285y4q5rpyjr4w93rdnx38wrhinxnw";
+  };
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  installPhase = ''
+    mkdir -p $out/bin
+    mkdir -p $out/lib
+    cp lib/tr-patcher-all.jar $out/lib/tr-patcher.jar
```
Comment (doronbehar) [inline pkgs/games/tr-patcher/default.nix:23]:
```suggestion
    install -Dm644 lib/tr-patcher-all.jar $out/lib/tr-patcher.jar
```

File: pkgs/games/tr-patcher/default.nix [hunk 27]
```diff
@@ -0,0 +1,28 @@
...
+    cp ${translation-file} $out/lib/Translation.txt
```
Comment (doronbehar) [inline pkgs/games/tr-patcher/default.nix:24]:
```suggestion
    install -Dm644 ${translation-file} $out/lib/Translation.txt
```

**Review by doronbehar — CHANGES_REQUESTED** [review]
Small nitpicks. Besides that all is good.

File: pkgs/development/python-modules/python-sat/default.nix (hunk 25 above)
Comment (doronbehar) [reply pkgs/development/python-modules/python-sat/default.nix:4]: For me, on Neovim, there's:

```vim
set listchars=tab:›\ ,trail:-,extends:»,precedes:«,eol:¬
```

File: pkgs/games/tr-patcher/default.nix [hunk 28]
```diff
@@ -0,0 +1,28 @@
+{ stdenv, fetchurl, fetchzip, jre, makeWrapper }:
```
Comment (jonringer) [inline pkgs/games/tr-patcher/default.nix:1]: NIT
```suggestion
{ stdenv, fetchurl, fetchzip, jre, makeWrapper }:

```

File: pkgs/games/tes3cmd/default.nix [hunk 29]
```diff
@@ -0,0 +1,20 @@
...
+    sha256 = "01zqplp8yb0xnl54963n0zkz66rf3hn2x3i255jlhdhx1c43jba7";
+  };
+
+  buildInputs = [ perlPackages.perl ];
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp tes3cmd $out/bin/tes3cmd
+  '';
+}
```
Comment (jonringer) [inline pkgs/games/tes3cmd/default.nix:20]: please add a meta section for tes3cmd

File: pkgs/development/python-modules/python-sat/default.nix (hunk 25 above)
Comment (marius851000) [reply pkgs/development/python-modules/python-sat/default.nix:4]: Thank. Never took care about this configuration in my editor. I've added the editorconfig plugin and enabled non visible character display to my editor.

**Comment by doronbehar** [comment]
A few issues with the executables:

```
$ ./results/portmod/bin/inquisitor
ERROR: Cannot find repository for the current directory. Please run from within the repository you wish to inspect
Traceback (most recent call last):
  File "/nix/store/wnp22084acnxvf8mb79ff04d7g8igzln-portmod-2.0_beta9/bin/.inquisitor-wrapped", line 9, in <module>
    sys.exit(main())
  File "/nix/store/wnp22084acnxvf8mb79ff04d7g8igzln-portmod-2.0_beta9/lib/python3.8/site-packages/portmod/inquisitor.py", line 200, in main
    REAL_ROOT = os.path.realpath(repo_root)
  File "/nix/store/346skv0d24rqnf4npknbp9h5bs14j8zy-python3-3.8.6/lib/python3.8/posixpath.py", line 390, in realpath
    filename = os.fspath(filename)
TypeError: expected str, bytes or os.PathLike object, not NoneType
```

```
$ ./results/portmod/bin/openmw-conflicts
Traceback (most recent call last):
  File "/nix/store/wnp22084acnxvf8mb79ff04d7g8igzln-portmod-2.0_beta9/bin/.openmw-conflicts-wrapped", line 9, in <module>
    sys.exit(main())
  File "/nix/store/wnp22084acnxvf8mb79ff04d7g8igzln-portmod-2.0_beta9/lib/python3.8/site-packages/portmod/openmw_conflicts.py", line 17, in main
    mod_dirs = get_vfs_dirs()
  File "/nix/store/wnp22084acnxvf8mb79ff04d7g8igzln-portmod-2.0_beta9/lib/python3.8/site-packages/portmod/vfs.py", line 111, in get_vfs_dirs
    return read_list(os.path.join(env.MOD_DIR, "vfs"))
  File "/nix/store/wnp22084acnxvf8mb79ff04d7g8igzln-portmod-2.0_beta9/lib/python3.8/site-packages/portmod/repo/list.py", line 12, in read_list
    with open(listpath, mode="r") as list_file:
FileNotFoundError: [Errno 2] No such file or directory: '/home/doron/.local/share/portmod/mods/vfs'
```

All other executables from new packages in this PR are fine as far as `--help` is concerned. Is this an issue @marius851000 ? or is this an expected behavior due to not found paths?

**Comment by marius851000** [comment]
inquisitor need to be run inside a package repository (there is one at https://gitlab.com/portmod/openmw-mods )

I have no idea what the second did, but it worked well on my side. I indeed have a ~/.local/share/portmod/mods/vfs . I suppose that can be solve by installing a package.
